imageregistry/v1/types.go

```diff
@@ -268,6 +268,32 @@ type ImageRegistryConfigStorageAzure struct {
 	CloudName string `json:"cloudName,omitempty"`
 }
 
+// ImageRegistryConfigStorageIBMCOS holds the information to configure
+// the registry to use IBM Cloud Object Storage for backend storage.
+type ImageRegistryConfigStorageIBMCOS struct {
+	// bucket is the bucket name in which you want to store the registry's
+	// data.
+	// Optional, will be generated if not provided.
+	// +optional
+	Bucket string `json:"bucket,omitempty"`
+	// location is the IBM Cloud location in which your bucket exists.
+	// Optional, will be set based on the installed IBM Cloud location.
+	// +optional
+	Location string `json:"location,omitempty"`
+	// resourceGroupName is the name of the IBM Cloud resource group that this
+	// bucket is associated with.
+	// +optional
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
Based on the crd property descriptions, only bucket and location are optional.
If that's true, shouldn't `omitempty` be removed here from the required properties?

Reply:
Actually, they are all optional because most of these values can be computed based on the cluster it's installed on. Add more to the descriptions.

Reply:
Why is `ResourceGroupName` present? Is it even possible to create a bucket in a rg that differs from the COS service instance?

Reply:
@BobbyRadford The `ResourceGroupName` here isn't meant to set a different resource group for service instance and bucket. But rather, if you need to create a COS instace in a different resource group than the OpenShift cluster.

Reply:
So `ResourceGroupName` is for the bucket _and_ the service instance?

Reply:
Correct